tests/unit/steps/test_steps_plan.py

```diff
@@ -761,57 +761,69 @@ async def test_get_upgradable_hypervisors_machines(
 
 
 @pytest.mark.parametrize("backup", [True, False])
-@patch("cou.steps.plan.UpgradePlan")
-@patch("cou.steps.plan.PreUpgradeStep")
-def test_generate_common_plan(mock_PreUpgradeStep, mock_UpgradePlan, backup, cli_args):
+def test_generate_common_plan(backup, cli_args):
     target = OpenStackRelease("victoria")
     nova_compute = MagicMock(spec_set=OpenStackApplication)()
-    mock_analysis_result = MagicMock(spec=Analysis)()
+
+    # analysis_result.model.wait_for_active_idle() should have __name__
+    # to not raise an error.
+    class AnalysisResult(MagicMock):
+        __name__ = "wait_for_active_idle"
+
+    mock_analysis_result = AnalysisResult(spec=Analysis)()
     mock_analysis_result.apps_control_plane = [nova_compute]
 
     cli_args.backup = backup
 
-    cou_plan.generate_common_plan(mock_analysis_result, cli_args, target)
+    plan = cou_plan.generate_common_plan(mock_analysis_result, cli_args, target)
 
     if backup:
-        assert mock_PreUpgradeStep.call_count == 2
+        assert len(plan.sub_steps) == 2
+        assert plan.sub_steps[1].description == "Backup mysql databases"
     else:
-        mock_PreUpgradeStep.call_count == 1
+        assert len(plan.sub_steps) == 1
+        assert plan.sub_steps[0].description != "Backup mysql databases"
```

Comment:
Like this you are testing two things at once, I do not understand what we want to test. I really think we should define exp_steps even in parametrize.